### PR TITLE
on-dhcp-proxy default ipxeboot

### PR DESCRIPTION
**Pull request by anhou**
resolve https://github.com/RackHD/RackHD/issues/104

related PR https://github.com/RackHD/on-http/pull/186

@RackHD/rackhd_dev @RackHD/corecommitters  @ytjohn

### CODE REVIEW

**Comment by ytjohn** [comment]
+1

File: spec/lib/message-handler-spec.js [hunk 1]
```diff
@@ -257,8 +264,40 @@ describe("MessageHandler", function() {
...
+            waterline.nodes.findByIdentifier.resolves(node);
+
+            return messageHandler.getKnownNodeAction('testnodeid')
+            .then(function(out) {
+                expect(out).to.have.property('action').that.equals('ignore');
+            });
+        });
+
+        it("should get the action for a known node if it does not have an active task" +
+                " and havs bootSettings", function() {
```
Comment (heckj) [inline spec/lib/message-handler-spec.js]: typo: `havs` should be `have`

**Comment by heckj** [comment]
@anhou +1 with a PR to the docs detailing what you have and how to set it up so that a node auto-PXE boots, even without a workflow defined. Also, let's make sure to sync this up with @brianparry so we can add "bootSettings" as a first class object with validation to the V2 API so that we don't run into the same issue of confusing, undocumented behavior that we saw with obmSettings in the V1.1 API.

With this capability, we should have an explicit resource or way to update it with validatable tests, beyond the sort of dynamic "hack it onto the object model" that we've been getting away with using PATCH to date...

**Comment by anhou** [comment]
@heckj Good points, I'll have a PR for docs

**Comment by yyscamper** [comment]
:+1:

**Pull request merged**